src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { getMovie } from '../../api';
+import { MovieCard } from '../MovieCard';
+
+type Props = {
+  addMovie: (movie: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [hasError, setHasError] = useState(false);
+  const [isMovieFetching, setIsMovieFetching] = useState(false);
+  const submitButtonText = movie ? 'Search Again' : 'Find a movie';
+
+  const getMovieFromResponse = (response: MovieData | ResponseError) => {
+    if ('Error' in response) {
+      setHasError(true);
+
+      return null;
+    }
+
+    const imgUrl = response.Poster === 'N/A'
+      ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+      : response.Poster;
+    const imdbUrl = `https://www.imdb.com/title/${response.imdbID}`;
+
+    const selectedMovie: Movie = {
+      title: response.Title,
+      description: response.Plot,
+      imgUrl,
+      imdbUrl,
+      imdbId: response.imdbID,
+    };
+
+    setHasError(false);
+
+    return selectedMovie;
+  };
+
+  const getMovieByName = async (event: { preventDefault: () => void; }) => {
```

Comment:
Add correct type for events 
```suggestion
  const getMovieByName = async (event: React.MouseEvent<HTMLButtonElement>) => {
```